Editor/EditorCore/FreeFormBoxTool.cs

```diff
@@ -0,0 +1,225 @@
+                    m_Modifications.Add(pbmesh, new InternalModification(pbmesh.positionsInternal));
+                }
+            }
+        }
+
+        /// <summary>
+        ///   <para>Use this method to implement a custom editor tool.</para>
+        /// </summary>
+        /// <param name="window">The window that is displaying the custom editor tool.</param>
+        public override void OnToolGUI(EditorWindow window)
+        {
+            base.OnToolGUI(window);
+
+            foreach (var obj in targets)
+            {
+                var pbmesh = obj as ProBuilderMesh;
+
+                if (pbmesh != null)
+                {
+                    if(m_BoundsHandleActive && GUIUtility.hotControl == k_HotControlNone)
+                        EndBoundsEditing();
+
+                    if(Mathf.Approximately(pbmesh.transform.lossyScale.sqrMagnitude, 0f))
+                        return;
+
+                    DoManipulationGUI(pbmesh);
+                }
+            }
+        }
+
+        protected override void DoManipulationGUI(Object toolTarget)
+        {
+            ProBuilderMesh mesh = toolTarget as ProBuilderMesh;
+            if(mesh == null)
+                return;
+
+            var matrix = IsEditing
+                ? m_ActiveBoundsState.positionAndRotationMatrix
+                : Matrix4x4.TRS(mesh.transform.position, mesh.transform.rotation, Vector3.one);
+
+            using (new Handles.DrawingScope(matrix))
+            {
+                m_BoundsHandle.SetColor(m_BoundsHandleColor);
+
+                CopyColliderPropertiesToHandle(mesh.transform, mesh.mesh.bounds);
+
+                EditorGUI.BeginChangeCheck();
+                m_BoundsHandle.DrawHandle();
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    BeginBoundsEditing(mesh);
+                    CopyHandlePropertiesToCollider(mesh);
+                }
+
+                DoRotateHandlesGUI(toolTarget, mesh, mesh.mesh.bounds);
+            }
+        }
+
+        protected override void UpdateTargetRotation(Object toolTarget, Quaternion rotation)
+        {
+            ProBuilderMesh mesh = toolTarget as ProBuilderMesh;
+            if(mesh == null)
+                return;
+
+            if ( rotation.Equals(Quaternion.identity) )
+                return;
+
+            InternalModification currentModification = m_Modifications[mesh];
+            currentModification.rotation = rotation * currentModification.rotation;
+
+            Bounds bounds = mesh.mesh.bounds;
+
+            var origVerts = new Vector3[mesh.vertexCount] ;
+            Array.Copy(currentModification.vertices, origVerts, mesh.vertexCount);
+
+            for (int i = 0; i < origVerts.Length; ++i)
+                origVerts[i] = currentModification.rotation * origVerts[i] + bounds.center;
+
+            mesh.mesh.vertices = origVerts;
+            mesh.ReplaceVertices(origVerts);
```

Comment:
why are we copying the vertex array, then re-assigning it? this should just access `mesh.positionsInternal`